api/v1alpha1/validation.go

```diff
@@ -69,6 +69,8 @@ func (r DeviceSpec) Validate(fleetTemplate bool) []error {
 		allErrs = append(allErrs, paramErrs...)
 		if !containsParams {
 			allErrs = append(allErrs, validation.ValidateOciImageReference(&r.Os.Image, "spec.os.image")...)
+		} else {
```

Comment:
would it make the flow simpler if we encapsulated `validateOciImageReferenceWithTemplates` and `validateParametersInString` inside ValidateOciImageReference, and added the isFleetTemplate bool to the validateOciImageReference signature?